project2/py/transform.py

```diff
@@ -0,0 +1,144 @@
+import utils
+from recipe import Recipe
+from fractions import Fraction
+
+
+HEALTHY_TRANSFORMATION_FACTOR = Fraction(0.75)
+UNHEALTHY_TRANSFORMATION_FACTOR = Fraction(1.5)
+
+
+def to_vegetarian(recipe, kb):
+    substituter = make_name_substituter(kb.get_meat_substitute)
+
+    new_recipe = substitute_ingredients(
+        recipe, kb.is_meat, substituter)
+    return new_recipe
+
+
+def from_vegetarian(recipe, kb):
+    substituter = make_name_substituter(kb.get_meat)
+
+    new_recipe = substitute_ingredients(
+        recipe, kb.is_vegge_protein, substituter)
+
+    return new_recipe
+
+
+def to_healthy(recipe, kb):
+    new_recipe = change_quantities(
+        recipe,
+        kb.is_scalable_ingredient,
+        HEALTHY_TRANSFORMATION_FACTOR)
+
+    substituter = make_name_substituter(kb.get_unhealthy_substitute)
+
+    new_recipe = substitute_ingredients(
+        new_recipe,
+        kb.is_unhealthy,
+        substituter
+    )
+
+    return new_recipe
+
+
+def from_healthy(recipe, kb):
+    new_recipe = change_quantities(
+        recipe,
+        kb.is_scalable_ingredient,
+        UNHEALTHY_TRANSFORMATION_FACTOR)
+
+    substituter = make_name_substituter(kb.get_healthy_substitute)
+
+    new_recipe = substitute_ingredients(
+        new_recipe,
+        kb.is_healthy,
+        substituter
+    )
+
+    return new_recipe
+
+
+def scale_quantity(ingredient, factor):
+    """
+        scales quantity of ingredient by factor
+    """
+    new_quantity = ingredient.quantity * factor
+    new_ingredient = ingredient.substitute_quantity(new_quantity)
+    return new_ingredient
+
+
+def change_quantities(recipe, predicate, factor):
+    """
```

Comment:
are these comments too or strings?

Reply:
this is a python multi-line comment, they're convention for function documentation in python